### PR TITLE
Use `snaps@0.35.2-flask.1` and `snaps@1.0.0-prerelease.1`

### PR DESCRIPTION
**Pull request by FrederikBolding**
## Explanation

Bumps Snaps Flask packages to `0.35.2-flask.1` and handles any breaking changes. This PR also bumps the Snaps stable packages to `1.0.0-prerelease.1`, this allows us to remove some temporary mocks that were added for a breaking change that had not been released on the Snaps audit branch.

Due to changes in the build output in the Snaps packages this PR also changes some imports slightly and therefore introduces more of the Snaps dependencies in the MetaMask build.

### CODE REVIEW

**Comment by FrederikBolding** [comment]
@SocketSecurity ignore @metamask/providers@11.0.0
@SocketSecurity ignore @metamask/snaps-controllers@0.35.2-flask.1
@SocketSecurity ignore @metamask/snaps-utils@0.35.2-flask.1

File: builds.yml [hunk 1]
```diff
@@ -69,7 +69,7 @@ buildTypes:
       - SEGMENT_FLASK_WRITE_KEY
       - ALLOW_LOCAL_SNAPS: true
       - REQUIRE_SNAPS_ALLOWLIST: false
-      - IFRAME_EXECUTION_ENVIRONMENT_URL: https://execution.metamask.io/0.16.1-flask.1/index.html
+      - IFRAME_EXECUTION_ENVIRONMENT_URL: https://execution.metamask.io/0.35.0-flask.1/index.html
```
Comment (Mrtenz) [inline builds.yml:72]:
```suggestion
      - IFRAME_EXECUTION_ENVIRONMENT_URL: https://execution.metamask.io/0.35.2-flask.1/index.html
```

File: test/env.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
 process.env.METAMASK_ENVIRONMENT = 'test';
 process.env.SUPPORT_LINK = 'https://support.metamask.io';
 process.env.IFRAME_EXECUTION_ENVIRONMENT_URL =
-  'https://execution.metamask.io/0.16.0-flask.1/index.html';
+  'https://execution.metamask.io/0.35.0-flask.1/index.html';
```
Comment (Mrtenz) [inline test/env.js:4]:
```suggestion
  'https://execution.metamask.io/0.35.2-flask.1/index.html';
```

**Review by Mrtenz — CHANGES_REQUESTED** [review]
These should be updated once deployment is approved.

**Review by Mrtenz — APPROVED** [review]

**Review by hmalik88 — APPROVED** [review]

**Pull request merged**